src/js/node/events.js

```diff
@@ -107,8 +109,12 @@ const emitWithRejectionCapture = function emit(type, ...args) {
   if (events === undefined) return false;
   var handlers = events[type];
   if (handlers === undefined) return false;
-  for (var handler of [...handlers]) {
-    var result = handler.apply(this, args);
+  // iterate in reverse so the handlers array does not need to be cloned
+  for (let i = handlers.length - 1; i >= 0; --i) {
+    let result;
+    // for performance reasons Function.call(...) is used if only one arg needs to be passed
+    if (args.length === 1) result = handlers[i].call(this, args[0]);
+    else result = handlers[i].apply(this, args);
```

Comment:
It would be wonderful if event loops didn't have the `this` binding. I experimented over the years with trying to detect if it was needed but the detection was always too heavy. Essentially array-function or bound listeners don't need the `.call` and could be executed straight up.

Reply:
... oh wait. bound functions and array functions don't have a `.prototype` property.... 
```js
var a = () => {}
a.prototype
// undefined
'prototype' in a
// false
var a = function() {}.bind();
a.prototype
// undefined
'prototype' in a
// false
```
Maybe a little sniff there could turn it into a non-call invocation.

Reply:
I found that there was actually no measurable difference between `handler()` and `handler.call(this)` so I just kept the `.call` version for now. Getting rid of `.apply` made a huge difference though.

Reply:
Perfect! Thank you for digging into that.